pkgs/development/tools/parsing/tree-sitter/default.nix

```diff
@@ -58,32 +38,34 @@ let
 
 in rustPlatform.buildRustPackage {
   pname = "tree-sitter";
-  inherit version;
-  inherit src;
+  inherit src version cargoSha256;
 
-  patches = [
-    # the web ui requires tree-sitter compiled to js and wasm
-    ./disable-web-ui.patch
-  ];
+  nativeBuildInputs = [ emscripten which ];
 
   postPatch = ''
     # needed for the tests
     rm -rf test/fixtures/grammars
     ln -s ${grammars} test/fixtures/grammars
   '';
 
+  # Compile web assembly with emscripten. The --debug flag prevents us from
+  # minifying the JavaScript; passing it allows us to side-step more Node
+  # JS dependencies for installation.
+  preBuild = ''
+    HOME=/tmp
+    bash ./script/build-wasm --debug
+  '';
```

Comment:
We now support web assembly and can remove our old patch (which didn't apply cleanly) and vendoring hacks.  Someone who wants to go really above and beyond could also get the JS minify step working, but I'm satisfied taking it to this point.

Reply:
Let’s add an option to make the web-ui optional.